### PR TITLE
Project structure section.

### PR DESCRIPTION
**Pull request by zebganzo**
Expand the project structure section with a suggestion in case of complex projects.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -130,6 +130,9 @@ To keep all those hundreds of source files from ending up in the same directory,
 
 First, create them as groups (little yellow "folders") within the group with your project's name in Xcode's Project Navigator. Then, for each of the groups, link them to an actual directory in your project path by opening their File Inspector on the right, hitting the little gray folder icon, and creating a new subfolder with the name of the group in your project directory.
 
+However, in case of a complex project, splitting the files based on the features is a good idea to keep in mind. 
+Some instances of the components mentioned before usually shared with most of the code of the project, but the main part of them are only related to a particular feature. So, makes sense to organise them under the same folder named with the specific feature.
```
Comment (jpeinelt) [inline README.md]: ... are usualley shared...
..., but most of them ....

Also would it make sense to mention the new behaviour in Xcode9 where groups are stronger coupled with the filesystem/folders. So creating a group now creates also a folder, so the paragraph in line 131 is kinda out of date?

**Review by jpeinelt — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (boris-tschirschwitz) [reply README.md]: Yes, I think this advice can be deleted now, that Xcode does the right thing. Maybe it still makes sense to point out that older code bases might still need to be updated.

File: README.md [hunk 2]
```diff
@@ -130,6 +130,9 @@ To keep all those hundreds of source files from ending up in the same directory,
 
 First, create them as groups (little yellow "folders") within the group with your project's name in Xcode's Project Navigator. Then, for each of the groups, link them to an actual directory in your project path by opening their File Inspector on the right, hitting the little gray folder icon, and creating a new subfolder with the name of the group in your project directory.
 
+However, in case of a complex project, splitting the files based on the features is a good idea to keep in mind.
```
Comment (boris-tschirschwitz) [inline README.md]: Maybe we should be more opinionated here, stating that once it makes sense to endow ones project tree with more depth one should cluster by feature, not by function, as in a well structured project separate features should be as loosely coupled as possible, while coupling between one component's feature is natural.

File: README.md (hunk 1 above)
Comment (boris-tschirschwitz) [inline README.md]: So, __it__ makes sense to organise… (Not sure if the comma is needed).

**Review by boris-tschirschwitz — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (boris-tschirschwitz) [reply README.md]: Actually, jpmcglone already opened a well written issue about that.

File: README.md (hunk 1 above)
Comment (erwald) [reply README.md]: I agree that we can delete this now.

File: README.md (hunk 2 above)
Comment (erwald) [inline README.md]: "However, for complex projects, ..."
"splitting the files by feature is a good idea."
Maybe?

**Review by erwald — APPROVED** [review]
Approved but have a look at the comments before merging. 👍

**Pull request closed without merging**